src/System.IO.FileSystem/src/System/IO/FileSystemInfo.cs

```diff
@@ -8,11 +8,13 @@
 using Microsoft.Win32;
 using System.Text;
 using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
 using System.Runtime.Versioning;
 
 namespace System.IO
 {
-    public abstract partial class FileSystemInfo
+    [Serializable]
```

Comment:
Yes.  For a type to be serializable, it must be attributed as [Serializable]; just implementing ISerializable isn't sufficient.  A formatter first checks whether the type is [Serializable], and then it looks for other things on the type (e.g. ISerializable) to influence how it does the serialization.